sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/AstyanaxDataReaderDAO.java

```diff
@@ -980,6 +1009,17 @@ public Record apply(Row<ByteBuffer, UUID> row) {
             protected MultiTableScanResult computeNext() {
                 while (_iter.hasNext()) {
                     Row<ByteBuffer, UUID> row = _iter.next();
+                    ColumnList<UUID> rowColumns = row.getColumns();
+
+                    // filter the columns of the row if a cutOff time is specified.
+                    if (timeUUID != null) {
+                        Iterables.removeIf(rowColumns, new Predicate<Column<UUID>>() {
+                            @Override
+                            public boolean apply(@Nullable Column<UUID> column) {
+                                return (TimeUUIDs.compare(column.getUUIDValue(), timeUUID) >= 0);
+                            }
+                        });
+                    }
```

Comment:
Has this been tested?  Reviewing the code it doesn't look like `iterator.remove()` is supported for the instance returned from `ColumnList.iterator()`.  Maybe use `Iterables.filter()` instead.

An alternate solution would be to change the column range to end on the first possible UUID with the provided timestamp.